src/lib_rules.ml

```diff
@@ -340,7 +341,7 @@ module Gen (P : Install_rules.Params) = struct
 
     let dep_graphs = Ocamldep.rules cctx in
 
-    let dynlink = lib.dynlink in
+    let dynlink = lib.dynlink && ctx.supports_shared_libraries in
```

Comment:
This pattern (`lib.dynlink` && `ctx.supports_shared_libraries`) appears 3 times. Is it worth extracting a combinator for it? (I guess `Context` would be the natural place for it)

Reply:
I don't think it would be much clearer than this though

Reply:
Yes, but I'm wondering whether we'll think of adding `&& ctx.supports_shared_libraries` next time we'll use `lib.dynlink`. The current solution works for me, though.

Reply:
I guess we could add an API like this:

```ocaml
module Dynlink_supported : sig
  type t
  val of_bool : bool -> t
  module Global : sig
    type t
    val of_bool : bool -> t
  end
  val get : t -> Global.t -> bool
end
```

and change the type of `lib.dynlink` to `Dynlink_supported.t` and the type of `ctx.supports_shared_libraries` to `Dynlink_supported.Global.t`. Then we are sure to catch such errors.

Reply:
I did that